core/nullDC.cpp

```diff
@@ -40,35 +40,41 @@ settings_t settings;
 #include <windows.h>
 #endif
 
-int GetFile(char *szFileName, char *szParse=0, u32 flags=0)
+int GetFile(std::string& diskFileName)
 {
-	cfgLoadStr("config","image",szFileName,"null");
-	if (strcmp(szFileName,"null")==0)
+	cfgLoadStr("config", "image", diskFileName, "");
+#if HOST_OS==OS_WINDOWS
+	if (diskFileName.empty())
 	{
-	#if HOST_OS==OS_WINDOWS
+		diskFileName.resize(MAX_PATH);
 		OPENFILENAME ofn;
-		ZeroMemory( &ofn , sizeof( ofn));
-	ofn.lStructSize = sizeof ( ofn );
-	ofn.hwndOwner = NULL  ;
-	ofn.lpstrFile = szFileName ;
-	ofn.lpstrFile[0] = '\0';
-	ofn.nMaxFile = MAX_PATH;
-	ofn.lpstrFilter = "All\0*.*\0\0";
-	ofn.nFilterIndex =1;
-	ofn.lpstrFileTitle = NULL ;
-	ofn.nMaxFileTitle = 0 ;
-	ofn.lpstrInitialDir=NULL ;
-	ofn.Flags = OFN_PATHMUSTEXIST|OFN_FILEMUSTEXIST ;
-
-		if (GetOpenFileNameA(&ofn))
+		ZeroMemory(&ofn, sizeof(ofn));
+		ofn.lStructSize = sizeof(ofn);
+		ofn.lpstrFile = &diskFileName[0];
+		ofn.nMaxFile = MAX_PATH;
+		ofn.lpstrFilter = "Supported Disk Image Files(.CDI, .GDI)\0*.CDI;*.GDI\0\0";
```

Comment:
We also support *.CHD.
The order should be ``"Supported Disk Image Files (*.GDI, *.CHD, *.CDI)\0*.GDI;*.CHD;*.CDI\0\0";``